api/src/middleware/__tests__/assessmentsRouter.ts

```diff
@@ -59,17 +81,520 @@ describe('assessmentsRouter', () => {
+    it("should show a facilitator the full submission information for a participant's ungraded submitted assessment, including the correct answers", done => {
+      const facilitatorPrincipalId = 3;
+      const participantPrincipalId = 2;
+      const programAssessmentId = 1;
+      const submissionId = 1;
+      const curriculumAssessmentId = 1;
+      const curriculumAssessment = {
+        id: curriculumAssessmentId,
+        title: 'Assignment 1: React',
+        description: 'Your assignment for week 1 learning.',
+        max_score: 10,
+        max_num_submissions: 1,
+        time_limit: 120,
+        curriculum_id: 3,
+        activity_id: 97,
+        principal_id: facilitatorPrincipalId,
+        questions: [
+          {
+            id: 1,
+            assessment_id: curriculumAssessmentId,
+            title: 'What is React?',
+            description: '',
+            question_type: 'single choice',
+            answers: [
+              {
+                id: 1,
+                question_id: 1,
+                title: 'A relational database management system',
+                description: '',
+                sort_order: 1,
+                correct_answer: true,
+              },
+            ],
+            correct_answer_id: 1,
+            max_score: 1,
+            sort_order: 1,
+          },
+        ],
+      };
+      const programAssessment = {
+        id: programAssessmentId,
+        program_id: 1,
+        assessment_id: curriculumAssessmentId,
+        available_after: '2023-02-06',
+        due_date: '2023-02-10',
+      };
+
+      const assessmentSubmission = {
+        id: 2,
+        assessment_id: programAssessmentId,
+        principal_id: participantPrincipalId,
+        assessment_submission_state: 'Submitted',
+        opened_at: '2023-02-09 12:00:00',
+        submitted_at: '2023-02-09 13:23:45',
+        responses: [
+          {
+            id: 1,
+            answer_id: 1,
+            assessment_id: 1,
+            submission_id: 2,
+            question_id: 1,
+          },
+        ],
+      };
+
+      const response: SubmittedAssessment = {
+        curriculum_assessment: curriculumAssessment,
+        program_assessment: programAssessment,
+        submission: assessmentSubmission,
+      };
+
+      mockGetProgramIdByProgramAssessmentId.mockResolvedValue([
+        { program_id: programAssessment.program_id },
+      ]);
+      mockFindRoleInProgram.mockResolvedValue({ title: 'facilitator' });
+      mockProgramAssessmentById.mockResolvedValue([programAssessment]);
+      mockGetCurriculumAssessmentById.mockResolvedValue(curriculumAssessment);
+      mockSubmissionDetails.mockResolvedValue([assessmentSubmission]);
+      mockPrincipalId(facilitatorPrincipalId);
+      appAgent
+        .get(`/${programAssessmentId}/submissions/${submissionId}`)
+
+        .expect(200, itemEnvelope(response), err => {
+          expect(mockProgramAssessmentById).toHaveBeenCalledWith(
```

Comment:
Your `expect(...).toHaveBeenCalledWith(...)` function calls should match in both number and order with your `mock....mockResolvedValue()` function calls in every test.